includes/class-wc-beta-tester-import-export.php

```diff
@@ -0,0 +1,211 @@
+				esc_html_e( 'Import WooCommerce Settings', 'woocommerce-beta-tester' );
+				?></button>
+				<input type="file" name="<?php echo static::IMPORT_FILENAME; ?>" />
+			</form>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Export settings in json format.
+	 */
+	public function export_settings() {
+		if ( ! wp_verify_nonce( $_REQUEST['_wpnonce'], static::NONCE_ACTION ) ) {
+			header( 'HTTP/1.1 403 Forbidden' );
+			exit;
+		}
+
+		$filename = sprintf( 'woocommerce-settings-%s.json', gmdate( 'Ymdgi' ) );
+		wc_set_time_limit( 0 );
+		wc_nocache_headers();
+		header( 'Content-Type: text/csv; charset=utf-8' );
+		header( 'Content-Disposition: attachment; filename=' . $filename );
+		header( 'Pragma: no-cache' );
+		header( 'Expires: 0' );
+		echo wp_json_encode( $this->get_settings() );
+		exit;
+	}
+
+	/**
+	 * Import settings in json format if submitted.
+	 */
+	public function maybe_import_settings() {
+		if ( empty( $_POST ) || empty( $_POST['action'] ) || $_POST['action'] !== static::IMPORT_ACTION ) {
+			return;
+		}
+
+		if ( ! wp_verify_nonce( $_POST['_wpnonce'], static::NONCE_ACTION ) ) {
+			$this->add_message( __( 'Invalid submission', 'woocommerce-beta-tester' ) );
+			return;
+		}
+
+		if ( ! empty( $_FILES[ static::IMPORT_FILENAME ] ) ) {
+			$tmp_file = $_FILES[ static::IMPORT_FILENAME ]['tmp_name'];
+			if ( empty( $tmp_file ) ) {
+				$this->add_message( __( 'No file uploaded.', 'woocommerce-beta-tester' ) );
+				return;
+			}
+
+			if ( is_readable( $tmp_file ) ) {
+				$maybe_json = file_get_contents( $tmp_file );
+				$settings = json_decode( $maybe_json, true );
+				if ( $settings !== null ) {
+					foreach ( $this->get_setting_list() as $option_name ) {
+						if ( ! isset( $settings[ $option_name ] ) ) {
+							continue;
+						}
+						$setting = maybe_unserialize( $settings[ $option_name ] );
+						if ( is_null( $setting ) ) {
+							delete_option( $option_name );
+						} else {
+							update_option( $option_name, $setting );
+						}
+					}
+					$this->add_message( __( 'Settings Imported', 'woocommerce-beta-tester' ), 'updated' );
+				} else {
+					$this->add_message( __( 'File did not contain well formed JSON.', 'woocommerce-beta-tester' ) );
+				}
+			} else {
+				$this->add_message( __( 'Fie could not be read.', 'woocommerce-beta-tester' ) );
+			}
+		} else {
+			$this->add_message( __( 'No file uploaded.', 'woocommerce-beta-tester' ) );
+		}
+	}
+
+	/**
+	 * Get an array of the WooCommerce related settings.
+	 */
+	protected function get_settings() {
+		$settings = array();
+		if ( current_user_can( 'manage_woocommerce' ) ) {
+			foreach ( $this->get_setting_list() as $option_name ) {
+				$setting = get_option( $option_name );
+				if ( false === $setting ) {
+					$setting = null;
+				}
+				$settings[ $option_name ] = is_string( $setting ) ? $setting : serialize( $setting );
+			}
+		}
+		return $settings;
+	}
+
+	/**
+	 * Add an setttings import status message.
```

Comment:
Super tiny nitpick: `a` instead of `an`.

Reply:
There were also 3 `t`s in settings :)